libs/shared/types/src/lib/types.ts

```diff
@@ -6,9 +6,74 @@ export enum EventStatus {
   ERROR = 'ERROR',
 }
 
-export type S3Source = string;
+export enum InputType {
+  MP4_FILE = 'MP4_FILE',
+  TS_FILE = 'TS_FILE',
+  URL_PULL = 'URL_PULL',
+  RTP_PUSH = 'RTP_PUSH',
+  RTMP_PUSH = 'RTMP_PUSH',
+  RTMP_PULL = 'RTMP_PULL',
+  MEDIACONNECT = 'MEDIACONNECT',
+  SRT_CALLER = 'SRT_CALLER',
+}
+
+export type S3Source = { inputType: InputType.MP4_FILE; value: string };
+
+export type TsFile = { inputType: InputType.TS_FILE; value: string };
```

Comment:
Could we make sure all those types end with `Source`?

`TsFileSource`, `HlsSource`, etc...